litgpt/__main__.py

```diff
@@ -0,0 +1,132 @@
+    from litgpt.generate.adapter_v2 import main as generate_adapter_v2_fn
+    from litgpt.generate.sequentially import main as generate_sequentially_fn
+    from litgpt.generate.tp import main as generate_tp_fn
+    from litgpt.scripts.convert_pretrained_checkpoint import (
+        convert_pretrained_checkpoint as convert_pretrained_checkpoint_fn,
+    )
+    from litgpt.scripts.convert_hf_checkpoint import convert_hf_checkpoint as convert_hf_checkpoint_fn
+    from litgpt.scripts.convert_lit_checkpoint import convert_lit_checkpoint as convert_lit_checkpoint_fn
+
+    parser_data = {
+        "pretrain": {"help": "Pretrain a model.", "fn": pretrain_fn},
+        "finetune": {
+            "help": "Finetune a model with one of our existing methods.",
+            "lora": {"help": "Finetune a model with LoRA.", "fn": finetune_lora_fn},
+            "full": {"help": "Finetune a model.", "fn": finetune_full_fn},
+            "adapter": {"help": "Finetune a model with Adapter.", "fn": finetune_adapter_fn},
+            "adapter_v2": {"help": "Finetune a model with Adapter v2.", "fn": finetune_adapter_v2_fn},
+        },
+        "generate": {
+            "help": "Generate text samples based on a model and tokenizer.",
+            "base": {"fn": generate_base_fn, "help": "Default generation option."},
+            "full": {"fn": generate_full_fn, "help": "For models finetuned with `litgpt finetune full`."},
+            "lora": {"fn": generate_lora_fn, "help": "For models finetuned with `litgpt finetune lora`."},
+            "adapter": {"fn": generate_adapter_fn, "help": "For models finetuned with `litgpt finetune adapter`."},
+            "adapter_v2": {
+                "fn": generate_adapter_v2_fn,
+                "help": "For models finetuned with `litgpt finetune adapter v2`.",
+            },
+            "sequentially": {
+                "fn": generate_sequentially_fn,
+                "help": "Generation script that partitions layers across devices to be run sequentially.",
+            },
+            "tp": {
+                "fn": generate_tp_fn,
+                "help": "Generation script that uses tensor parallelism to run across devices.",
+            },
+        },
+        "chat": {"help": "Chat with a model.", "fn": chat_fn},
+        "convert": {
+            "help": "Utilities to convert from and to LitGPT.",
+            "to_litgpt": {"fn": convert_hf_checkpoint_fn, "help": "Convert Hugging Face weights to LitGPT weights."},
+            "from_litgpt": {"fn": convert_lit_checkpoint_fn, "help": "Convert LitGPT weights to Hugging Face weights."},
+            "pretrained_checkpoint": {
+                "fn": convert_pretrained_checkpoint_fn,
+                "help": "Convert a checkpoint after pretraining.",
+            },
+        },
+        "download": {"help": "Download weights or tokenizer data from the Hugging Face Hub.", "fn": download_fn},
+        # TODO: should this be at the first level?
```

Comment:
Question for the audience. Suggestion for different name or organization is also welcome

Reply:
> TODO: should this be at the first level?

For now having `merge_lora` should be good IMO. The "merge" command might be used for something else in the future.